### ISSUE TITLE
Windows 10 Windows Update Display Problems

### ISSUE THREAD

**Issue by Elshara** [nvaccess/nvda#5970]
In windows 10, windows update does not work well with NVDA at all. IT is next to impossible to know how to interact with it, whether to know something is checked when a new update appears, how to install it, where to go to see previous updates. I'm not sure this is a Microsoft issue.

**Comment by PratikP1** [comment]
Can you specify what version of Windows 10 you are using? Are you running an insider build?

**Comment by jcsteh** [comment]
You can use NVDA's review cursor and object navigation to access 
information in Windows Update.

**Comment by josephsl** [comment]
Hi, As Jamie said, you can use object navigation to read info regarding Windows Update status in Windows 10. Apart from that, without any info as to which version of Windows 10 you’ve got, NVDA version and what not, we cannot proceed (also, please be ready to respond to questions from others, as it is important to get the dialogue going). Thanks.

**Comment by Elshara** [comment]
I'm using the latest build of windows insider.

**Comment by PratikP1** [comment]
I cannot verify this problem with the latest insider build. Object navigation allows reading.

**Comment by Elshara** [comment]
Even if insert b causes the entire window to freeze half way through
describing it as the settings window. The only settings window that
causes this with no link support?

**Comment by PratikP1** [comment]
Can you provide an exact description? Perhaps the exact set of steps that will need to be taken to see the problem that you’re  seeing.

**Comment by Elshara** [comment]
The problems start when a new update is ready to be installed. There's
no description of what the update is it's just a link. When you click
on it, no information is given to check or uncheck it as in if it
appeared in a list like it used to in windows 8 and earlier, now it
doesn't. It just shows up as a link. No way to know if it is checked
or if there is an install link.

**Comment by PratikP1** [comment]
I highly recommend using object navigation  to navigate the Windows update dialog. It is true that the old style checkboxes do not exist. Updates are listed in a list box and a details link does appear. A request has been logged with Microsoft regarding checkboxes.

**Comment by Elshara** [comment]
This is the same reason why I could never use the windows store to
download apps or play with them because virtually nothing on it,
including Cortana, is in any way accessible. Hopefully this has
changed recently.

**Comment by PratikP1** [comment]
Windows store was a different matter. It was based on Edge, which, with the first release of Windows 10, was not fully accessible. It’s beginning to change with every inside builds.

**Comment by derekriemer** [comment]
And windows 10 is not possible to forgo individual updates in fact windows forces you to update. I think this may be why you are not seeing checkboxes (it is simply a list of updates with links to explain more about each of them). I know this is frustrating, but it's a decision that Microsoft has made it and are not turning around as far as I know.just another one of those weird things about windows 10 that I can decide.

Sent from my heavily encrypted iPhone.
Please disregard errors as this is a smaller device.

**Comment by Elshara** [comment]
I don't mind the forced updates. I mind that I can't even read about
them when I click on the link, the entire window freezes when NVDA
tries to read it. I like the fact that Microsoft makes you update. It
means better security and that everyone gets the same access to the
features that can be talked about in the Windows Feedback app. As for
the window, Edge is improving but it's got a long way to go and the
windows store is tricky to use. Try to install something like Skype
from it, or use the music player app for example. It's not like
regular navigation will get you through all the windows the app
supports.

**Comment by Brian1Gaff** [comment]
Can you imagine though, if you are for example on a charge by usage mobile 
connection or a pre paid for x gigs plan, what a large windows update will 
do to this?
 I can which is why all my computers which use such connections run 
windows7.


The daft bit of windows 10 updates seems to be that one can read about them 
after they are installed, but not before you click the button for optional 
ones or at all with the automatic ones.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by PratikP1** [comment]
I understand and agree with many points regarding the Windows update problem. However Microsoft has chosen to deliver updates the way they have for their own reasons. NVAccess cannot do anything about that. I respectfully suggest that this is not an NVDA bug.

**Comment by Elshara** [comment]
I understand what you are saying. I'm not sure I have much more to say
to this other than it's out of my hands.

**Comment by ehollig** [comment]
As @jcsteh said in https://github.com/nvaccess/nvda/issues/5970#issuecomment-220575872, "You can use NVDA's review cursor and object navigation to access
information in Windows Update." It unfortunately does not appear that Windows allows you to uncheck particular updates. CC @josephsl for any additional information.

**Comment by josephsl** [comment]
Hi, the UI presentation has seen some changes in Fall Creators Update (where each update item will present its own download/installation status), but as noted before, you cannot check individual updates to not be installed. Thanks.
